core/src/main/scala/org/apache/spark/storage/BlockManagerMessages.scala

```diff
@@ -97,6 +97,15 @@ private[spark] object BlockManagerMessages {
   case class GetLocationsAndStatus(blockId: BlockId, requesterHost: String)
     extends ToBlockManagerMaster
 
+  case class GetLocalDirs(executorIds: Array[String]) extends ToBlockManagerMaster
+
+  /**
+   * The response message of `GetLocalDirs` request.
+   *
+   * @param localDirs follows the order of the block manager IDs used in the `GetLocalDirs` request.
```

Comment:
Wouldn't it be better to return a map?

Reply:
Keeping the same ordering is chosen over a map because of the RPC message serialization.

Reply:
Why? Maps are just as serializable as arrays.

Reply:
I was wrong here: somehow I thought it is better to avoid maps in messages because of Kryo. But now I see it works perfectly (even with Kryo).